lib/improver/spotdata/site_data.py

```diff
@@ -0,0 +1,302 @@
+            self.site_ids[i_site] = '{:03d}{:03d}{:04d}'.format(
+                int(self.latitudes[i_site]+180),
+                int(self.longitudes[i_site]+180),
+                int(np.nanmax([self.altitudes[i_site], 0.])))
+
+        self.utc_offsets = np.full(n_sites, np.nan)
+        for i_site, site in enumerate(site_data):
+            if 'gmtoffset' in site.keys():
+                self.utc_offsets[i_site] = site['gmtoffset']
+            elif 'utcoffset' in site.keys():
+                self.utc_offsets[i_site] = site['utcoffset']
+            elif 'utc_offset' in site.keys():
+                self.utc_offsets[i_site] = site['utc_offset']
+
+            # If it's not been set, set it with the longitude based method.
+            if np.isnan(self.utc_offsets[i_site]):
+                self.utc_offsets[i_site], = set_utc_offset([site['longitude']])
+
+        return self.construct_site_dictionary()
+
+    def runtime_list(self, site_properties):
+        """
+        Use data provided on the command line/controlling suite at runtime.
+
+        Args:
+        -----
+        site_properties : dict
+            Contains:
+
+            latitudes/longitudes : lists of floats
+                Lists of coordinates at which to extract data, these define the
+                positions of the SpotData sites.
+
+            altitudes : list of floats
+                List of altitudes that can optionally be used in defining sites
+                on the fly. If unset the altitudes will be assumed as
+                equivalent to which ever neighbouring grid point is used as a
+                data source.
+
+            site_ids : list of integers
+                Optional integer site_ids that will be used as keys for the
+                site information in the returned site data dictionary. If unset
+                a simple counting index is used.
+
+        Returns:
+        --------
+        sites : dict
+            Restructured dictionary of site data.
+
+        Raises:
+        -------
+        ValueError : If runtime defined lists of site properties are of unequal
+        length, suggesting an error with the input (e.g. a site is missing
+        some required property).
+
+        """
+        n_sites = site_properties['latitudes'].__len__()
+
+        if site_properties['longitudes'].__len__() != n_sites:
+            raise ValueError('Unequal number of latitude and longitudes '
+                             'given for sites.')
+        if (('altitudes' in site_properties.keys() and
+             site_properties['altitudes'] is not None and
+             site_properties['altitudes'].__len__() != n_sites) or
+            ('site_ids' in site_properties.keys() and
+             site_properties['site_ids'] is not None and
+             site_properties['site_ids'].__len__() != n_sites)):
+            raise ValueError('Unequal number of altitudes/site_ids and '
+                             'latitude/longitudes given for sites.')
+
```

Comment:
You should be able to get away with
`'altitudes' in site_properties`
no need for .keys()
get rid of \_\_len\_\_

The ValueError should not have / in the message, this should be or. Unequal number of altitudes or site_id given for sites.
This should really be two checks but I'm guessing you are trying to reduce the number of unittest.